### PR TITLE
feat: split fees between treasury&author

### PR DESCRIPTION
**Pull request by weichweich**
# Pay Fees to Authors & Treasury (https://github.com/KILTprotocol/ticket/issues/1660)

* [x] 50:50 split needs to be confirmed
* enable transfers
* enable gratitude

[M](https://kilt-cast.monday.com/boards/1792114080/pulses/1846293585)[M](https://kilt-cast.monday.com/boards/1792114080/pulses/1792587139)[M](https://kilt-cast.monday.com/boards/1792114080/pulses/1807402622)

## Checklist:

- [ ] I have verified that the code works
  - [ ] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] This PR does not introduce new custom types
  - [ ] If not, I have opened a companion PR with the type changes in the [KILT types-definitions repository](https://github.com/KILTprotocol/type-definitions/pulls)
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: primitives/src/fees.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+where
+	R: pallet_balances::Config,
+	Beneficiary1: OnUnbalanced<NegativeImbalanceOf<R>>,
+	Beneficiary2: OnUnbalanced<NegativeImbalanceOf<R>>,
+	Ratio: Get<(u32, u32)>,
+{
+	fn on_unbalanceds<B>(mut fees_then_tips: impl Iterator<Item = NegativeImbalanceOf<R>>) {
+		let ratio = Ratio::get();
+		if let Some(fees) = fees_then_tips.next() {
+			// for fees, 80% to treasury, 20% to author
```
Comment (Diiaablo95) [inline primitives/src/fees.rs:44]:
```suggest
			// for fees, 50% to treasury, 50% to author
```

**Review by Diiaablo95 — APPROVED** [review]
LGMT! A minor comment mismatch, I guess.

File: primitives/src/fees.rs (hunk 1 above)
Comment (weichweich) [reply primitives/src/fees.rs:44]: we don't know the ratio here. So I would delete that line

File: primitives/src/fees.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+use crate::{constants::MILLI_KILT, Balance};
+use frame_support::{
+	traits::{Currency, Get, Imbalance, OnUnbalanced},
+	weights::{DispatchClass, WeightToFeeCoefficient, WeightToFeeCoefficients, WeightToFeePolynomial},
+};
+use pallet_balances::WeightInfo;
+use smallvec::smallvec;
+use sp_runtime::Perbill;
+
+use crate::{AccountId, NegativeImbalanceOf};
```
[the remark this answers is not in the document]
Comment (weichweich) [reply primitives/src/fees.rs]: 628c62dbd9d670b3cb5217fe0b58424ee5760462

File: primitives/src/fees.rs (hunk 1 above)
Comment (weichweich) [reply primitives/src/fees.rs]: 628c62dbd9d670b3cb5217fe0b58424ee5760462

**Review by wischli — APPROVED** [review]
LGTM!

C: Add unit tests, at least for fee splitting [similar to as is done in Polkadot](https://github.com/paritytech/polkadot/blob/master/runtime/common/src/impls.rs#L193).

**Pull request merged**